docs/proposals/20220712-cluster-api-addon-orchestration.md

```diff
@@ -0,0 +1,448 @@
+	ClusterSelector ClusterSelectorLabel `json:"clusterSelector"`
+
+	// ChartName is the name of the Helm chart in the repository.
+	ChartName string `json:"chartName,omitempty"`
+
+	// RepoURL is the URL of the Helm chart repository.
+	RepoURL string `json:"repoURL,omitempty"`
+
+	// ReleaseName is the release name of the installed Helm chart. If it is not specified, a name will be generated.
+	// +optional
+	ReleaseName string `json:"releaseName,omitempty"`
+
+	// Version is the version of the Helm chart. If it is not specified, the latest version will be used.
+	// +optional
+	Version string `json:"version,omitempty"`
+
+	// Namespace is the namespace the Helm release will be installed on each selected Cluster. If it is not specified, the default namespace.
+	// +optional
+	Namespace string `json:"namespace,omitempty"`
+
+	// Values is an inline YAML representing the values for the Helm chart. This YAML supports Go templating to reference
+	// fields from each selected workload Cluster and programatically create and set values.
+	// +optional
+	Values string `json:"values,omitempty"`
+}
+
+// ClusterSelectorLabel defines a key/value pair used to select Clusters with a label matching the specified key and value.
+type ClusterSelectorLabel struct {
+	// Key is the label key.
+	Key string `json:"key"`
+
+	// Value is the label value.
+	Value string `json:"value"`
+}
+
+// HelmChartProxyStatus defines the observed state of HelmChartProxy.
+type HelmChartProxyStatus struct {
+	// Ready is true when the HelmReleaseProxySpec for each selected Cluster is up to date.
+	// +optional
+	Ready bool `json:"ready"`
+
+	// MatchingClusters is the list of references to Clusters selected by the ClusterSelectorLabel.
+	// +optional
+	MatchingClusters []corev1.ObjectReference `json:"matchingClusters"`
+
+	// Conditions defines current state of the HelmChartProxy.
+	// +optional
+	Conditions clusterv1.Conditions `json:"conditions,omitempty"`
+
+	// FailureReason will be set in the event that there is a an error reconciling the HelmChartProxy.
+	// +optional
+	FailureReason string `json:"failureReason,omitempty"`
+}
+```
+
+#### HelmReleaseProxy CRD
+
+The second CRD, HelmReleaseProxy, is responsible for installing and orchestrating the Helm release on a selected workload Cluster. The HelmReleaseProxy is created as part of the parent HelmChartProxy flow. The parent HelmChartProxy will be responsible for keeping its HelmReleaseProxy children up-to-date as its lifecycle is updated (for example, if the `values` properties of a HelmChartProxy are updated, its child HelmReleaseProxy resources will be updated as well).
+
+```go
+// HelmReleaseProxySpec defines the desired state of HelmReleaseProxy.
+type HelmReleaseProxySpec struct {
+	// ClusterRef is a reference to the Cluster to install the Helm release on.
+	ClusterRef *corev1.ObjectReference `json:"clusterRef,omitempty"`
```

Comment:
is this cross namespace reference or should this be a local / local typed ref?

Reply:
(somewhat related to my point above: are those CRDs namespaced or cluster-wide)

Reply:
Currently there's nothing to restrict it to the same namespace. Would it be beneficial to restrict it to the same namespace?

Reply:
I'm +1 to have this namespaced for starting. It is simpler and we are consistent with everything else in CAPI.

Reply:
Sounds good. TBH I'm a bit on the fence about it but I agree that we can move forward with making it namespaced for simplicity and consistency with CAPI.